api/submariner/v1alpha1/submariner_types.go

```diff
@@ -31,38 +31,46 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type BrokerDetails struct {
+	BrokerK8sApiServer       string `json:"brokerK8sApiServer"`
+	BrokerK8sApiServerToken  string `json:"brokerK8sApiServerToken"`
+	BrokerK8sCA              string `json:"brokerK8sCA"`
+	BrokerK8sRemoteNamespace string `json:"brokerK8sRemoteNamespace"`
+	BrokerK8sInsecure        bool   `json:"brokerK8sInsecure,omitempty"`
+}
+
+type ClusterDetails struct {
+	ClusterID           string               `json:"clusterID"`
+	Namespace           string               `json:"namespace"`
+	Repository          string               `json:"repository,omitempty"`
+	ImageVersion        string               `json:"version,omitempty"`
+	SubmarinerDebug     bool                 `json:"debug"`
+	CoreDNSCustomConfig *CoreDNSCustomConfig `json:"coreDNSCustomConfig,omitempty"`
+	// +listType=set
+	CustomDomains       []string             `json:"customDomains,omitempty"`
+	ImageOverrides      map[string]string    `json:"imageOverrides,omitempty"`
+}
+
 // SubmarinerSpec defines the desired state of Submariner
 // +k8s:openapi-gen=true
 type SubmarinerSpec struct {
 	Broker                   string               `json:"broker"`
-	BrokerK8sApiServer       string               `json:"brokerK8sApiServer"`
-	BrokerK8sApiServerToken  string               `json:"brokerK8sApiServerToken"`
-	BrokerK8sCA              string               `json:"brokerK8sCA"`
-	BrokerK8sRemoteNamespace string               `json:"brokerK8sRemoteNamespace"`
+	BrokerDetails            BrokerDetails        `json:"broker_details"`
```

Comment:
Same as above, this is a breaking change, as is the `ClusterDetails` field below.